core/utils.py

```diff
@@ -64,37 +68,41 @@ def __init__(self, bot) -> None:
         self.cachedUsers = {}
 
     async def get_channel(self, id: int) -> TextChannel:
-        if id not in self.cachedChannels:
-
-            # "Clear the cache" if it gets too big
-            if len(self.cachedChannels) >= MAX_CACHE_LENGTH:
-                self.cachedChannels = {}
-
-            self.cachedChannels[id] = await self.bot.fetch_channel(id)
+        if self._is_data_fresh(self.cachedChannels, id):
+            response = await self.bot.fetch_channel(id)
+            response.expires = utc_now() + CACHE_EXPIRY_SECONDS
+            self.cachedChannels[id] = response
 
         return self.cachedChannels[id]
 
     async def get_message(self, channel: TextChannel, id: int) -> Message:
-        if id not in self.cachedMessages:
-
-            # "Clear the cache" if it gets too big
-            if len(self.cachedMessages) >= MAX_CACHE_LENGTH:
-                self.cachedMessages = {}
-
-            self.cachedMessages[id] = await channel.fetch_message(id)
+        if self._is_data_fresh(self.cachedMessages, id):
+            response = await channel.fetch_message(id)
+            response.expires = utc_now() + CACHE_EXPIRY_SECONDS
+            self.cachedMessages[id] = response
 
         return self.cachedMessages[id]
 
     async def get_user(self, id: int) -> User:
-        if id not in self.cachedUsers:
+        if self._is_data_fresh(self.cachedUsers, id):
+            response = await self.bot.fetch_user(id)
+            response.expires = utc_now() + CACHE_EXPIRY_SECONDS
+            self.cachedUsers[id] = response
 
-            # "Clear the cache" if it gets too big
-            if len(self.cachedUsers) >= MAX_CACHE_LENGTH:
-                self.cachedUsers = {}
+        return self.cachedUsers[id]
 
-            self.cachedUsers[id] = await self.bot.fetch_user(id)
+    async def _is_data_fresh(self, cache: Dict[any], id: int) -> None:
+        if len(cache) >= MAX_CACHE_LENGTH:
+            cache = {}
```

Comment:
Could we cycle the oldest value out of the cache if it reaches the maximum size, rather than clearing the whole thing?